react/__tests__/FilterNavigator.test.js

```diff
@@ -45,15 +51,405 @@ describe('<FilterNavigator />', () => {
+        to: '/fulltextterm/eletronicos',
+        query: 'map=ft',
+      })
+    )
+  })
+
+  it('should not allow to remove a root category when there is a children selected', () => {
+    const tree = JSON.parse(JSON.stringify(categoriesTree))
+
+    const selectedDepartment = tree.find(
+      (department) => department.value === 'eletronicos'
+    )
+
+    selectedDepartment.selected = true
+
+    selectedDepartment.children.find(
+      (category) => category.value === 'smartphones'
+    ).selected = true
+
+    const { getByText } = renderComponent({
+      query: 'eletronicos/smartphones',
+      tree,
+    })
+
+    fireEvent.click(getByText(/Eletrônicos/))
+
+    expect(mockNavigate).not.toHaveBeenCalled()
+  })
+
+  it('should allow to remove a root category when it is a fulltext search', () => {
+    const tree = JSON.parse(JSON.stringify(categoriesTree))
+
+    const selectedDepartment = tree.find(
+      (department) => department.value === 'eletronicos'
+    )
+
+    selectedDepartment.selected = true
+
+    const { getByText } = renderComponent({
+      query: 'fulltextterm',
+      map: 'ft',
+      tree,
+    })
+
+    fireEvent.click(getByText(/Eletrônicos/))
+
+    expect(mockNavigate).toHaveBeenCalledTimes(1)
+    expect(mockNavigate).toHaveBeenCalledWith(
+      expect.objectContaining({
+        to: '/fulltextterm',
+      })
+    )
+  })
+
+  it('should add unselected specification on click', () => {
+    const { getByTestId } = renderComponent({
+      specificationFilters: specifications,
+      filters: specifications,
+      tree: [],
+    })
+
+    fireEvent.click(getByTestId('check-specification-color-pink'))
+
+    expect(mockNavigate).toHaveBeenCalledTimes(1)
+    expect(mockNavigate).toHaveBeenCalledWith(
+      expect.objectContaining({
+        to: '/pink',
+        query: 'map=color',
+      })
+    )
+  })
+
+  it('should remove selected specification on click', () => {
+    const filters = JSON.parse(JSON.stringify(specifications))
+
+    filters
+      .find((filter) => filter.title === 'store/search.filter.title.color')
+      .facets.find(
+        (filterOption) => filterOption.value === 'pink'
+      ).selected = true
+
+    const { getAllByTestId } = renderComponent({
+      query: 'top/pink',
+      map: 'ft,map',
+      specificationFilters: filters,
+      filters,
+      tree: [],
+    })
+
+    const [checkBox] = getAllByTestId('check-specification-color-pink')
+
+    fireEvent.click(checkBox)
+
+    expect(mockNavigate).toHaveBeenCalledTimes(1)
+    expect(mockNavigate).toHaveBeenCalledWith(
+      expect.objectContaining({
+        to: '/top',
+        query: 'map=ft',
+      })
+    )
+  })
+
+  it('should navigate to root category on key down', () => {
+    const { getByTestId } = renderComponent({
+      query: 'fulltextterm',
+      map: 'ft',
+    })
+
+    fireEvent.keyDown(getByTestId('root-category-eletronicos'), {
+      key: 'Enter',
+      code: 'Enetr',
```

Comment:
```suggestion
      code: 'Enter',
```
there are some more like this below